solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -499,6 +495,7 @@ public void doFilter(ServletRequest _request, ServletResponse _response, FilterC
         if (log.isDebugEnabled()) {
           log.debug("User principal: {}", request.getUserPrincipal());
         }
+        span.setTag(Tags.DB_USER, String.valueOf(request.getUserPrincipal()));
```

Comment:
This line is not reached if there is no auth enabled, and thus there will be no `db.user` tag set.

Reply:
Would it be more interesting to specify that the user is not authenticated via a value of None for example instead of bypassing this setting?

Reply:
If Auth is enabled but the user isn't authenticated, getUserPrincipal() will return null and then String.valueOf will yield "null".  I don't think we should add this header if there is no auth because the header is then inapplicable.